packages/common/src/components/FilterGroup/matchers.ts

```diff
@@ -96,6 +96,11 @@ const groupedEnumMatcher = {
   matchValue: enumMatcher.matchValue,
 };
 
+const dateMatcher = {
+  filterType: 'date',
+  matchValue: freetextMatcher.matchValue,
```

Comment:
ok, so this is not a "from date" but an "at date" filter ... but you need to adjust for timezone and UTC ...

a way to do it is to get the datetime of 00 and 24 of the day, maybe using a helper method:
```ts
/**
 * Given an ISO date string, returns two epoch timestamps: 
 * one for the start (00:00:00 UTC) and one for the end (23:59:59 UTC) of the UTC day of the date in the given ISO string.
 */
function getStartAndEndEpochs(isoDateString: string): [number, number] {
  const date = new Date(isoDateString);

  date.setUTCHours(0, 0, 0, 0);
  const startEpoch = date.getTime();

  date.setUTCHours(23, 59, 59, 999);
  const endEpoch = date.getTime();

  return [startEpoch, endEpoch];
}
```
and then check the date you have is between this timestamps
```ts
/**
 * Checks if a given date string falls within the start and end epoch times of another provided date string.
 * 
 * @param {string} referenceIsoDate - The reference ISO date string to determine the UTC day.
 * @param {string} targetIsoDate - The ISO date string to check if it falls within the determined UTC day.
 * @returns {boolean} - True if the targetIsoDate falls within the UTC day of the referenceIsoDate, false otherwise.
 */
function isDateWithinDay(referenceIsoDate: string, targetIsoDate: string): boolean {
  const [startEpoch, endEpoch] = getStartAndEndEpochs(referenceIsoDate);
  const targetEpoch = new Date(targetIsoDate).getTime();
  return targetEpoch >= startEpoch && targetEpoch <= endEpoch;
}

```

Reply:
used luxon lib to perform the calculations

Reply:
Your implementation with loxun.hasSame addresses the "is it this day" scenario effectively. However, it might not cater to the broader requirements of handling date ranges and date-time ranges. My initial example using the internal Date object's getTime was to demonstrate an approach that can be easily extended for both single day checks and date range evaluations. Could we consider adapting the implementation to ensure it's versatile for all these use cases?

about using external the library luxon, see:
https://github.com/kubev2v/forklift-console-plugin/pull/747#issuecomment-1757479621

Reply:
> Your implementation with loxun.hasSame addresses the "is it this day" scenario effectively. However, it might not cater to the broader requirements of handling date ranges and date-time ranges.
>

Please check the matcher implementation in PR #754 :
https://github.com/rszwajko/forklift-console-plugin/blob/7f99ef49ccafedd287790fe45cd96231a612d8bf/packages/common/src/components/FilterGroup/matchers.ts#L116

The line demonstrates parsing ISO time internal format and checking if date belongs to the interval in one line.

Links:
https://moment.github.io/luxon/api-docs/index.html#intervalfromiso
https://en.wikipedia.org/wiki/ISO_8601#Time_intervals

> My initial example using the internal Date object's getTime was to demonstrate an approach that can be easily extended for both single day checks and date range evaluations. 
>

Please create a PR with your code and describe what is the added value, what functionality is missing in Luxon and why we should switch.

Reply:
> Please create a PR with your code and describe what is the added value, what functionality is missing in Luxon and why we should switch.

I appreciate your suggestion. However, our initial agreement was to proceed using the internal date object. It would be helpful if we could stick to that plan and move forward as discussed. Thank you for understanding.